semgrep-core/src/analyzing/Dataflow_svalue.ml

```diff
@@ -51,10 +53,77 @@ let warning tok s =
 
 let str_of_name name = spf "%s:%d" (fst name.ident) name.sid
 
+let str_of_resolved_name name =
+  let name = Common.map fst name in
+  String.concat "." name
+
 (*****************************************************************************)
 (* Constness *)
 (*****************************************************************************)
 
+let hook_constness_table_of_functions = ref None
+
+let result_of_function_call_is_constant lang f args =
+  let check_f f_name =
+    match !hook_constness_table_of_functions with
+    | Some constness_f -> (
+        match constness_f f_name with
+        | Some Constant ->
+            logger#info "%s is always constant" f_name;
```

Comment:
Better use `logger#trace` or else the output of `semgrep --debug` becomes very large, and most of the times this info won't be of much use. Same for the other two uses of `logger#info` below.